### PR TITLE
Fix: A single-file app can have CRLF and random YAML documents order 

### PR DESCRIPTION
**Pull request by silvin-lubecki**
**- What I did**

Fix CRLF detection on single-file app (https://github.com/docker/app/issues/553)
Fix YAML documents order detection for single-file apps (https://github.com/docker/app/issues/249)

- Split and Merge remembers the CRLF detection while outputting the result

**- How to verify it**
Add e2e tests: 
- split/merge a single file with CRLF
- render a single file with documents in random order

**- Description for the changelog**
* Fix CRLF detection on single-file app
* Fix YAML documents order detection for single-file apps

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 
 import (
+	"bytes"
 	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"strings"
+
+	"github.com/docker/cli/cli/compose/loader"
```
Comment (thaJeztah) [inline loader/loader.go]: Looks like this one should be grouped with the ones below

**Comment by thaJeztah** [comment]
Did a quick check what the YAML spec says about newlines; https://yaml.org/spec/1.0/#id2565848

> Inside all scalar nodes, a compliant YAML processor must translate the two-character combination CR LF, any CR that is not followed by an LF, and any NEL into a single LF (this does not apply to escaped characters). LS and PS characters are preserved. These rules are compatible with Unicode's newline guidelines.

**Comment by silvin-lubecki** [comment]
Yes but I used the `1.2` version of the specs https://yaml.org/spec/1.2/spec.html#id2774608 says `CRLF` is a valid line break:
> YAML recognizes the following ASCII line break characters.
> [24]    b-line-feed    ::=    #xA    /* LF \*/    
> [25]    b-carriage-return    ::=    #xD    /* CR \*/    
> [26]    b-char    ::=    b-line-feed | b-carriage-return

**Comment by silvin-lubecki** [comment]
@jcsirot @rumpl PTAL

File: loader/loader.go (hunk 1 above)
Comment (silvin-lubecki) [reply loader/loader.go]: Fixed 👍

File: loader/loader.go [hunk 2]
```diff
@@ -1,38 +1,100 @@
...
+	nbLf := bytes.Count(data, lf)
+	switch {
+	case nbCrlf == nbLf:
+		// document contains only CRLF
+		return true
+	case nbCrlf == 0:
+		// document does not contain any CRLF
+		return false
+	default:
+		// document contains mixed line breaks, so let use the OS preferency
```
Comment (chris-crone) [inline loader/loader.go]: `// document contains mixed line breaks, so let use the OS preferency`

`// document contains mixed line breaks so use the OS default`

**Comment by silvin-lubecki** [comment]
PTAL @chris-crone

**Pull request merged**